Source/Core/DolphinQt2/Settings/AudioPane.cpp

```diff
@@ -0,0 +1,229 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/Settings/AudioPane.h"
+
+#include <QCheckBox>
+#include <QComboBox>
+#include <QGridLayout>
+#include <QGroupBox>
+#include <QLabel>
+#include <QRadioButton>
+#include <QSlider>
+#include <QSpinBox>
+#include <QVBoxLayout>
+
+#include "AudioCommon/AudioCommon.h"
+#include "DolphinQt2/Config/SettingsWindow.h"
+#include "DolphinQt2/Settings.h"
+
+AudioPane::AudioPane()
+{
+  CreateWidgets();
+  LoadSettings();
+  ConnectWidgets();
+
+  connect(&Settings::Instance(), &Settings::VolumeChanged, this, &AudioPane::OnVolumeChanged);
+}
+
+void AudioPane::CreateWidgets()
+{
+  auto* dsp_box = new QGroupBox(tr("DSP Emulator Engine"));
+  auto* dsp_layout = new QVBoxLayout;
+  dsp_box->setLayout(dsp_layout);
+  m_dsp_hle = new QRadioButton(tr("DSP HLE emulation (fast)"));
+  m_dsp_lle = new QRadioButton(tr("DSP LLE recompiler"));
+  m_dsp_interpreter = new QRadioButton(tr("DSP LLE interpreter (slow)"));
+
+  dsp_layout->addWidget(m_dsp_hle);
+  dsp_layout->addWidget(m_dsp_lle);
+  dsp_layout->addWidget(m_dsp_interpreter);
+
+  auto* volume_box = new QGroupBox(tr("Volume"));
+  auto* volume_layout = new QGridLayout;
+  m_volume_slider = new QSlider;
+  m_volume_indicator = new QLabel(tr("??? %"));
+
+  volume_box->setLayout(volume_layout);
+
+  m_volume_slider->setMinimum(0);
+  m_volume_slider->setMaximum(100);
+
+  m_volume_indicator->setAlignment(Qt::AlignVCenter | Qt::AlignHCenter);
+
+  volume_layout->addWidget(m_volume_slider, 1, 2);
+  volume_layout->addWidget(m_volume_indicator, 2, 1, 1, 3);
+
+  auto* backend_box = new QGroupBox(tr("Backend Settings"));
+  auto* backend_layout = new QGridLayout;
```

Comment:
With `QFormLayout`, this looks very odd on macOS because the default label alignment is on the right, but the checkbox is flush with the left side of the group box:

<img width="414" alt="screen shot 2017-06-21 at 3 16 17 pm copy" src="https://user-images.githubusercontent.com/594093/27409177-01389e9e-5695-11e7-90ec-544c2ad804a5.png">

One recommendation I found (from KDE) is to add labels for all checkboxes, though that then triggers a centered alignment for the entire layout under macOS:

<img width="414" alt="screen shot 2017-06-21 at 3 16 17 pm" src="https://user-images.githubusercontent.com/594093/27409108-9abbe5c2-5694-11e7-83b1-637008493d96.png">

This does look pretty neat, but is inconsistent with the rest of Dolphin's UI.

I'd suggest reverting this back to the QGridLayout for now, to match DolphinWX, and push the conversation of alignment to a later, non-code-focussed PR.